### ISSUE TITLE
DPE Inspector: Changing Terrain Queries and World Position values in Terrain World Debugger component crashes the Editor

### ISSUE THREAD

**Issue by LB-MateuszSzweda** [o3de/o3de#16721]
**Describe the bug**
Modifying values of queries and World Position found in _Terrain Queries_ dropdown of the **Terrain World Debugger** component crashes the Editor if `ed_enableDPE` flag is enabled.

**Call stack trace**

```
25) 00007FFBCC6BDEA7 (KERNELBASE) : UnhandledExceptionFilter
24) 00007FFBCBC66A06 (InProcessClient64) : (function-name not available)
23) 00007FFBCBC66A98 (InProcessClient64) : (function-name not available)
22) 00007FFBCBC66961 (InProcessClient64) : (function-name not available)
21) 00007FFBCEA754F0 (ntdll) : memset
20) 00007FFBCEA5C8B6 (ntdll) : _C_specific_handler
19) 00007FFBCEA723DF (ntdll) : _chkstk
18) 00007FFBCEA214B4 (ntdll) : RtlRaiseException
17) 00007FFBCEA70EEE (ntdll) : KiUserExceptionDispatcher
16) 00007FFB960DD635 (Qt5Core) : QVariant::cmp
15) 00007FFB806BD30E (Qt5Widgets) : QAbstractSpinBox::selectAll
14) 00007FFB806BE091 (Qt5Widgets) : QAbstractSpinBox::stepBy
13) 00007FFB806BEF8D (Qt5Widgets) : QAbstractSpinBox::wheelEvent
12) 00007FFB8057793E (Qt5Widgets) : QWidget::event
11) 00007FFB805549A1 (Qt5Widgets) : QApplicationPrivate::notify_helper
10) 00007FFB80553A23 (Qt5Widgets) : QApplication::notify
 9) 00007FFB960A17CB (Qt5Core) : QCoreApplication::notifyInternal2
 8) C:\o3de\Code\Framework\AzQtComponents\AzQtComponents\Components\GlobalEventFilter.cpp (73) : AzQtComponents::GlobalEventFilter::eventFilter
 7) C:\o3de\Code\Editor\Core\QtEditorApplication.cpp (140) : `anonymous namespace'::EditorGlobalEventFilter::eventFilter
 6) 00007FFB960A3764 (Qt5Core) : QCoreApplicationPrivate::sendThroughApplicationEventFilters
 5) 00007FFB805548E3 (Qt5Widgets) : QApplicationPrivate::notify_helper
 4) 00007FFB80553A23 (Qt5Widgets) : QApplication::notify
 3) 00007FFB960A17CB (Qt5Core) : QCoreApplication::notifyInternal2
 2) 00007FFB7CC375F5 (Qt5Gui) : QGuiApplicationPrivate::processWheelEvent
 1) 00007FFB7CC1EE99 (Qt5Gui) : QWindowSystemInterface::sendWindowSystemEvents
```

The Issue occurs on both Windows and Linux platforms.

For further information please refer to the attached logs and video file.

**Steps to reproduce**
1. Launch the Editor and load any level.
2. Enable DPE in Editor by inputting _ed_enableDPE=true_ command in the Console.
3. Add **Terrain World** and **Terrain World Debugger** components to the level prefab in the Entity Outliner.
4. Switch on the _Show Terrain Queries_ toggle in the **Terrain World Debugger** component.
5. Modify the World Position values or any of queries sliders for ex. Point count.

**Expected behavior**
The values are changed and the Editor remains stable.

**Actual behavior**
The Editor crashes upon changing Terrain Queries or World Position values.

**Screenshots/Video**

https://github.com/o3de/o3de/assets/86953456/248d9b78-ae2a-4dff-8c3e-507cf62fbf57

**Found in Branch**
Stabilization/2310

**Commit ID from** [o3de/o3de](https://github.com/o3de/o3de) **Repository**
[372b2b8](https://github.com/o3de/o3de/commit/372b2b829a767a8b522b05acc4b27810dbf49ad7)

**Desktop/Device:**
- Device: PC
- OS: Windows
- Version: 10 Pro
- CPU: AMD Ryzen 9 3900X
- GPU: NVIDIA GeForce RTX 2060 SUPER
- Memory: 16GB

**Additional context**
[error.zip](https://github.com/o3de/o3de/files/12619602/error.zip)
[error.log](https://github.com/o3de/o3de/files/12619605/error.log)
[Editor.log](https://github.com/o3de/o3de/files/12619607/Editor.log)

**Comment by cgalvan** [comment]
This has the same root cause as https://github.com/o3de/o3de/issues/16722

**Comment by LB-MateuszSzweda** [comment]
Issue verified as fixed on the Stabilization/2310 (d380ffa) branch.
